reconcile/test/test_utils_state.py

```diff
@@ -0,0 +1,65 @@
+import os
+from tempfile import NamedTemporaryFile
+
+import pytest
+import boto3
+from moto import mock_s3
+from reconcile.utils.state import State
+
+
+@pytest.fixture
+def accounts():
+    """Account name is needed to instantiate a State"""
+    return [{'name': 'some-account'}]
+
+
+@pytest.fixture
+def aws_credentials():
+    """Mocked AWS Credentials for moto."""
+    os.environ['AWS_ACCESS_KEY_ID'] = 'testing'
```

Comment:
I don't think these direct `os.environ` modifications are working as you expect they would.  Consider this:

```
def test_set_os_environ():
    os.environ['SOME_VAR'] = 'this-is-set'
    assert os.environ['SOME_VAR'] == 'this-is-set'

def test_get_os_environ():
    assert os.environ['SOME_VAR'] == 'this-is-set'
```

These environment variables persist beyond the test your're setting them in.  You might consider `patch.dict` and patching `os.environ` directly instead.  I'll only comment on this once, but this applies to all usage of `os.environ` in this PR.

Reply:
I've done some tests with [Monkeypatch](https://docs.pytest.org/en/6.2.x/monkeypatch.html#monkeypatching-environment-variables) and worked fine. Gona push it.